### ISSUE TITLE
Errors when using table filters behind a proxy

### ISSUE THREAD

**Issue by mattmalcher** [simonw/datasette#1883]
Using datasette==0.63 table filters do not respect the `base_url` setting as described [here](https://docs.datasette.io/en/stable/deploying.html#running-datasette-behind-a-proxy)

To reproduce, go to:
https://datasette-apache-proxy-demo.datasette.io/prefix/fixtures/binary_data

Then use the table filter buttons.
The `/prefix/` is dropped, resulting in URL not found:
https://datasette-apache-proxy-demo.datasette.io/fixtures/binary_data?_sort=rowid&rowid__exact=1

**Comment by simonw** [comment]
Great catch, thanks!

**Comment by simonw** [comment]
If you view source on that page the HTML looks correct:
```html
<form class="filters" action="/prefix/fixtures/binary_data" method="get">
```
(I just added a test that confirms this too.)

But... it looks like the bug is in the redirection code. 

https://datasette-apache-proxy-demo.datasette.io/prefix/fixtures/binary_data?_filter_column=rowid&_filter_op=exact&_filter_value=1&_sort=rowid returns the following:

    location: /fixtures/binary_data?_sort=rowid&rowid__exact=1

Which is incorrect.

**Comment by simonw** [comment]
I tried adding this test but it passed! I expected it to fail:

```python
def test_base_url_affects_filter_redirects(app_client_base_url_prefix):
    response = app_client_base_url_prefix.get(
        "/prefix/fixtures/binary_data?_filter_column=rowid&_filter_op=exact&_filter_value=1&_sort=rowid"
    )
    assert response.status == 302
    assert (
        response.headers["location"]
        == "/prefix/fixtures/binary_data?_sort=rowid&rowid__exact=1"
    )
```

**Comment by simonw** [comment]
Relevant code: https://github.com/simonw/datasette/blob/bbaab3b38ec2ce5944239ffbe2dd53328df40fff/datasette/views/table.py#L227-L249

**Comment by simonw** [comment]
`path_with_added_args(request, redirect_params)` should be preserving the current path from the request.

**Comment by simonw** [comment]
Is there a chance that it's Apache that's messing with that `location:` header here, not Datasette?

https://github.com/simonw/datasette/blob/bbaab3b38ec2ce5944239ffbe2dd53328df40fff/demos/apache-proxy/000-default.conf#L7-L13

**Comment by simonw** [comment]
https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#proxypass includes this note:

> Normally, mod_proxy will canonicalise ProxyPassed URLs. But this may be incompatible with some backends, particularly those that make use of *PATH_INFO*. The optional *nocanon* keyword suppresses this and passes the URL path "raw" to the backend.

**Comment by simonw** [comment]
I modified that config file to have this line instead:
```
    ProxyPass /prefix/ http://127.0.0.1:8001/ nocanon
```
And then deployed it by running:

    flyctl deploy --build-arg DATASETTE_REF=main

This does NOT seem to have fixed the bug:

```
~ % curl -i 'https://datasette-apache-proxy-demo.datasette.io/prefix/fixtures/binary_data?_filter_column=rowid&_filter_op=exact&_filter_value=1&_sort=rowid'
HTTP/2 302 
date: Fri, 11 Nov 2022 06:40:01 GMT
server: Fly/b1863e2e7 (2022-11-09)
location: /fixtures/binary_data?_sort=rowid&rowid__exact=1
link: </fixtures/binary_data?_sort=rowid&rowid__exact=1>; rel=preload
content-type: text/plain
x-proxied-by: Apache2 Debian
via: 2 fly.io
fly-request-id: 01GHJPNCF51CJ626EWZEHK2CH9-sjc
```

**Comment by simonw** [comment]
https://datasette-apache-proxy-demo.datasette.io/prefix/-/asgi-scope is useful:

It confirms that `/prefix/` is nowhere to be seen in the incoming request data:

```
 'path': '/-/asgi-scope',
 'query_string': b'',
 'raw_path': b'/-/asgi-scope',
```

**Comment by simonw** [comment]
Modifying that test to the following does indeed cause a failure:
```python
def test_base_url_affects_filter_redirects(app_client_base_url_prefix):
    response = app_client_base_url_prefix.get(
        "/fixtures/binary_data?_filter_column=rowid&_filter_op=exact&_filter_value=1&_sort=rowid"
    )
    assert response.status == 302
    assert (
        response.headers["location"]
        == "/prefix/fixtures/binary_data?_sort=rowid&rowid__exact=1"
    )
```

**Comment by simonw** [comment]
This time deployed with:
```
cd demos/apache-proxy
fly deploy --build-arg DATASETTE_REF=8d9a957c6329d26cc1e417b5d6911640d74765eb
```
To ensure the exact commit with the fix.

And that fixed it!
```
% curl -i 'https://datasette-apache-proxy-demo.datasette.io/prefix/fixtures/binary_data?_filter_column=rowid&_filter_op=exact&_filter_value=1&_sort=rowid'
HTTP/2 302 
date: Fri, 11 Nov 2022 06:54:45 GMT
server: Fly/b1863e2e7 (2022-11-09)
location: /prefix/fixtures/binary_data?_sort=rowid&rowid__exact=1
link: </prefix/fixtures/binary_data?_sort=rowid&rowid__exact=1>; rel=preload
content-type: text/plain
x-proxied-by: Apache2 Debian
via: 2 fly.io
fly-request-id: 01GHJQGBSXBR7E53TY0EKMQ9PA-sjc
```

**Comment by simonw** [comment]
I released that fix in Datasette 0.63.1: https://docs.datasette.io/en/stable/changelog.html#v0-63-1
